lib/Doctrine/Common/Cache/EncryptingCacheDecorator.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Doctrine\Common\Cache;
+
+use InvalidArgumentException as IAE;
+use RuntimeException as RE;
+
+class EncryptingCacheDecorator implements Cache
+{
+    /** @var Cache */
+    private $decorated;
+    /** @var resource */
+    private $publicKey;
+    /** @var resource */
+    private $privateKey;
+
+    /**
+     * @param Cache $decorated
+     * @param resource $publicKey
+     * @param resource $privateKey
+     *
+     * @throws RE If OpenSSL extension not installed.
+     * @throws IAE If either key is not an OpenSSL Key resource.
+     */
+    public function __construct(Cache $decorated, $publicKey, $privateKey)
+    {
+        if (!extension_loaded('openssl')) {
+            throw new RE('The OpenSSL extension is required to use the'
+                . __CLASS__ . '. Please install OpenSSL to use this feature.');
+        }
+
+        foreach (array($publicKey, $privateKey) as $key) {
+            if (!is_resource($key) || 'OpenSSL key' !== get_resource_type($key)) {
```

Comment:
OpenSSL functions also accept a path to a pem certificate file when calling encryption and decryption functions. Supporting it here would make it much simpler to use the class (especially when using a DI container to manage the instantiation: it is much easier to make it pass a string to the class than to make it build resources). It would also make it much easier to handle the lifecycle of the class (as you would not need to free the key resource anymore to avoid leaking it, which is currently quite hard to do)